ext/cl_khr_extended_bit_ops.asciidoc

```diff
@@ -0,0 +1,143 @@
+// Copyright 2018-2021 The Khronos Group. This work is licensed under a
+// Creative Commons Attribution 4.0 International License; see
+// http://creativecommons.org/licenses/by/4.0/
+
+[[cl_khr_extended_bit_ops]]
+== Extended Bit Operations
+
+This extension adds OpenCL C functions for performing extended bit operations.
+Specifically, the following functions are added:
+
+* bitfield insert: insert bits from one source operand into another source operand.
+* bitfield extract: extract bits from a source operand, with sign- or zero-extension.
+* bit reverse: reverse the bits of a source operand.
+
+=== General Information
+
+==== Name Strings
+
+`cl_khr_extended_bit_ops`
+
+==== Version History
+
+[cols="1,1,3",options="header",]
+|====
+| *Date*     | *Version* | *Description*
+| 2021-04-22 | 1.0.0     | Initial version.
+|====
+
+==== Dependencies
+
+This extension is written against the OpenCL 3.0 C Language Specification and the OpenCL SPIR-V Environment Specification Version V3.0.6.
+
+This extension requires OpenCL 1.0.
+
+=== New OpenCL C Functions
+
+[source]
+----
+gentype bitfield_insert( gentype base, gentype insert, uint offset, uint count )
+igentype bitfield_extract_signed( gentype base, uint offset, uint count )
```

Comment:
`i` is typically used as a prefix where signedness doesn't matter. `sgentype` might be clearer.

Reply:
Hmm, looking at the OpenCL C spec I think it's the other way around:

* `sgentype` is used to represent a scalar type, for `clamp`, `min`, and `max`.
* `igentype` is used to represent a signed type, for a handful of relational functions.

I think this means that this extension spec is consistent with the rest of the OpenCL C spec?

Reply:
Unfortunately you're right :). Agree consistency is more important so no change it is.